carbonmark-data/components/pages/PageWithTabs/index.tsx

```diff
@@ -0,0 +1,158 @@
+    queryParams.get("tab") || props.tabs[0].key
+  );
+  /** By default, the first option of each widget is selected */
+  const getDetaultOptions = () => {
+    return props.tabs.map((tab) =>
+      tab.optionsList ? tab.optionsList.map((options) => options[0].value) : []
+    );
+  };
+  /** Stores all options selections */
+  const [optionKeys, setOptionKeys] = useState<Key[][]>(getDetaultOptions);
+
+  /** Compute Dynamic Options Lists. Lists that contains only options that will lead to a valid content */
+  function getTabsDynamicOptionsList() {
+    // For each tab
+    return props.tabs.map((tab, tabIndex) => {
+      // For each option list
+      if (tab.optionsList)
+        return tab.optionsList.map((options, widgetIndex) => {
+          // Check if there is content for that option
+          const newOptions: Options<string> = [];
+          options.forEach((option) => {
+            const keys = [...optionKeys[tabIndex]];
+            keys[widgetIndex] = option.value;
+            const key = keys.join("|");
+            if (tab.contents && tab.contents[key] !== undefined)
+              newOptions.push(option);
+          });
+          return newOptions;
+        });
+      else return [];
+    });
+  }
+
+  let tabsDynamicOptionsList: Options<string>[][] = getTabsDynamicOptionsList();
+
+  const router = useRouter();
+  const pathname = usePathname();
+
+  const handleChange = (_: React.SyntheticEvent, newTab: string) => {
+    router.push(`${pathname}?tab=${newTab}`);
+    setActiveTab(newTab);
+  };
+
+  // Get Tab from URL
+  useEffect(() => {
+    const queryKey = queryParams.get("tab");
+    if (queryKey != null && props.tabs.some((tab) => tab.key == queryKey)) {
+      setActiveTab(queryKey);
+    }
+  }, []);
+
+  useEffect(() => {
+    tabsDynamicOptionsList = getTabsDynamicOptionsList();
+  }, [optionKeys]);
+
+  // Handle Options change
+  function onOptionChange(tabIndex: number, widgetIndex: number) {
+    return (value: Key) => {
+      const newOptionKeys = [...optionKeys];
+      newOptionKeys[tabIndex][widgetIndex] = value;
+      setOptionKeys(newOptionKeys);
+    };
+  }
+
+  /** Returns the tab to display given the chosen options
+   If the the chart attribute is filled, it is that chart.
+   If the charts attribute if filled, it is the chart that is referenced by the key :
+   `$t{topOptionKey}|$t{bottomOptionKey}` if both options are provided
+   topOptionKey if only topOptionKey is provided
+   bottomOptionKey if only bottom
+   "default" if there are no charts for the given option
+   empty component if everything fails
+  */
+  function displayedTab(tabIndex: number): React.ReactNode {
+    const tab = props.tabs[tabIndex];
+    if (tab?.content) return tab.content;
+
+    if (tab?.contents) {
+      const displayedKey = optionKeys[tabIndex].join("|");
+      return tab.contents[displayedKey] || <></>;
+    }
+  }
+  return (
+    <>
+      <h1>{props.title}</h1>
```

Comment:
Please use the new PageHeader component here

```suggestion
      <PageHeader title={props.title} />
```